shell/common/engine.cc

```diff
@@ -281,38 +280,40 @@ void Engine::SetViewportMetrics(const ViewportMetrics& metrics) {
   viewport_metrics_ = metrics;
   runtime_controller_->SetViewportMetrics(viewport_metrics_);
   if (animator_) {
-    if (dimensions_changed)
+    if (dimensions_changed) {
       animator_->SetDimensionChangePending();
-    if (have_surface_)
+    }
+    if (have_surface_) {
       ScheduleFrame();
+    }
   }
 }
 
 void Engine::DispatchPlatformMessage(fml::RefPtr<PlatformMessage> message) {
-  if (message->channel() == kLifecycleChannel) {
-    if (HandleLifecyclePlatformMessage(message.get()))
+  std::string channel = message->channel();
+  if (channel == kLifecycleChannel) {
+    if (HandleLifecyclePlatformMessage(message.get())) {
       return;
-  } else if (message->channel() == kLocalizationChannel) {
-    if (HandleLocalizationPlatformMessage(message.get()))
+    }
+  } else if (channel == kLocalizationChannel) {
+    if (HandleLocalizationPlatformMessage(message.get())) {
       return;
-  } else if (message->channel() == kSettingsChannel) {
+    }
+  } else if (channel == kSettingsChannel) {
     HandleSettingsPlatformMessage(message.get());
     return;
+  } else if (channel == kNavigationChannel) {
+    // If there's no runtime_, we may still need to set the initial route.
+    HandleNavigationPlatformMessage(std::move(message));
+    return;
   }
 
   if (runtime_controller_->IsRootIsolateRunning() &&
       runtime_controller_->DispatchPlatformMessage(std::move(message))) {
     return;
   }
 
-  // If there's no runtime_, we may still need to set the initial route.
-  if (message->channel() == kNavigationChannel) {
```

Comment:
Moving the navigation channel check to the `else if` block above is now preventing the runtime_controller from receiving navigation messages. The back button on Android no longer works.